core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -527,12 +527,14 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
       }
     }
 
-    // New consumer always hasNext
     override def hasData = true
 
     override def receive() : BaseConsumerRecord = {
-      while (recordIter == null || !recordIter.hasNext)
+      if (recordIter == null || !recordIter.hasNext) {
         recordIter = consumer.poll(1000).iterator
+        if (!recordIter.hasNext)
+          throw new ConsumerTimeoutException
```

Comment:
That's true. Although the frequency of `ConsumerTimeoutException` is probably low enough that it doesn't matter much.